src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
-
-	private final VertexFormat format;
-
-	private final TextureAtlasSprite facadeTexture;
-
-	private static final Set<ResourceLocation> warnedFor = new HashSet<>();
-
-	FacadeBuilder( VertexFormat format, Function<ResourceLocation, TextureAtlasSprite> bakedTextureGetter )
-	{
-		this.format = format;
-		this.facadeTexture = bakedTextureGetter.apply( TEXTURE_FACADE );
-	}
-
-	static Collection<ResourceLocation> getTextures()
-	{
-		return Collections.singletonList( TEXTURE_FACADE );
-	}
-
-	void addFacades( BlockRenderLayer layer, Map<EnumFacing, FacadeRenderState> facadesState, List<AxisAlignedBB> partBoxes, Set<EnumFacing> sidesWithParts, long rand, List<BakedQuad> quads )
+	public static double THICK_THICKNESS = 2D / 16D;
+	public static double THIN_THICKNESS = 1D / 16D;
+
+	public static final AxisAlignedBB[] THICK_FACADE_BOXES = new AxisAlignedBB[] {
+			new AxisAlignedBB( 0.0, 0.0, 0.0, 1.0, THICK_THICKNESS, 1.0 ),
+			new AxisAlignedBB( 0.0, 1.0 - THICK_THICKNESS, 0.0, 1.0, 1.0, 1.0 ),
+			new AxisAlignedBB( 0.0, 0.0, 0.0, 1.0, 1.0, THICK_THICKNESS ),
+			new AxisAlignedBB( 0.0, 0.0, 1.0 - THICK_THICKNESS, 1.0, 1.0, 1.0 ),
+			new AxisAlignedBB( 0.0, 0.0, 0.0, THICK_THICKNESS, 1.0, 1.0 ),
+			new AxisAlignedBB( 1.0 - THICK_THICKNESS, 0.0, 0.0, 1.0, 1.0, 1.0 )
+	};
+
+	public static final AxisAlignedBB[] THIN_FACADE_BOXES = new AxisAlignedBB[] {
+			new AxisAlignedBB( 0.0, 0.0, 0.0, 1.0, THIN_THICKNESS, 1.0 ),
+			new AxisAlignedBB( 0.0, 1.0 - THIN_THICKNESS, 0.0, 1.0, 1.0, 1.0 ),
+			new AxisAlignedBB( 0.0, 0.0, 0.0, 1.0, 1.0, THIN_THICKNESS ),
+			new AxisAlignedBB( 0.0, 0.0, 1.0 - THIN_THICKNESS, 1.0, 1.0, 1.0 ),
+			new AxisAlignedBB( 0.0, 0.0, 0.0, THIN_THICKNESS, 1.0, 1.0 ),
+			new AxisAlignedBB( 1.0 - THIN_THICKNESS, 0.0, 0.0, 1.0, 1.0, 1.0 )
+	};
+
+	private ThreadLocal<BakedPipeline> pipelines = ThreadLocal.withInitial( () -> BakedPipeline.builder()//
+			.addElement( "clamper", QuadClamper.FACTORY )// Clamper is responsible for clamping the vertex to the bounds specified.
+			.addElement( "face_stripper", QuadFaceStripper.FACTORY )//Strips faces if they match a mask.
+			.addElement( "corner_kicker", QuadCornerKicker.FACTORY )//Kicks the edge inner corners in, solves Z fighting
+			.addElement( "interp", QuadReInterpolator.FACTORY )//Re-Interpolates the UV's for the quad.
+			.addElement( "tinter", QuadTinter.FACTORY, false )// Tints the quad if we need it to. Disabled by default.
+			.addElement( "transparent", QuadAlphaOverride.FACTORY, false, e -> e.setAlphaOverride( 0x4C / 255F ) )//Overrides the quad's alpha if we are forcing transparent facades.
+			.build()//
+	);
+	private ThreadLocal<Quad> collectors = ThreadLocal.withInitial( Quad::new );
+
+	public void buildFacadeQuads( BlockRenderLayer layer, CableBusRenderState renderState, long rand, List<BakedQuad> quads, Function<ResourceLocation, IBakedModel> modelLookup )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "buildFacadeQuads" is 36 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)